python/src/test/java/org/apache/zeppelin/python/PythonInterpreterTest.java

```diff
@@ -0,0 +1,221 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one or more
+* contributor license agreements.  See the NOTICE file distributed with
+* this work for additional information regarding copyright ownership.
+* The ASF licenses this file to You under the Apache License, Version 2.0
+* (the "License"); you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*  http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package org.apache.zeppelin.python;
+
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.mockito.Mockito.eq;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.anyString;
+
+import org.apache.zeppelin.interpreter.InterpreterResult;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.net.SocketAddress;
+import java.util.Properties;
+
+public class PythonInterpreterTest {
+
+  Logger logger = LoggerFactory.getLogger(PythonProcess.class);
+
+  public static final String ZEPPELIN_PYTHON = "zeppelin.python";
+  public static final String DEFAULT_ZEPPELIN_PYTHON = "python";
+
+  PythonInterpreter pythonInterpreter = null;
+  PythonProcess mockPythonProcess;
+  String cmdHistory;
+
+  @Before
+  public void beforeTest() {
+    cmdHistory = "";
+
+    /*Mock python process*/
+    mockPythonProcess = mock(PythonProcess.class);
+    when(mockPythonProcess.getPid()).thenReturn((long) 1);
+    try {
+      when(mockPythonProcess.sendAndGetResult(anyString())).thenAnswer(new Answer<String>() {
+        @Override
+        public String answer(InvocationOnMock invocationOnMock) throws Throwable {
+          Object[] args = invocationOnMock.getArguments();
+          String cmd = (String) args[0];
+          if (cmd != null) {
+            cmdHistory += cmd;
+            String[] lines = cmd.split("\\n");
+            String output = "";
+            for (int i = 0; i < lines.length; i++) {
+              output += ">>>" + lines[i];
+            }
+            return output;
+
+          } else return ">>>";
```

Comment:
This need to be updated to conform the [project's code conventions](https://github.com/apache/incubator-zeppelin/blob/master/CONTRIBUTING.md#code-convention)